lib/lf/base/comm.cc

```diff
@@ -0,0 +1,180 @@
+#include "comm.h"
+
+namespace lf::base {
+
+namespace comm {
+
+namespace variables {
+
+std::map<std::string, std::pair<bs::hold_any, std::string>> kGlobalVars;
+
+void ListVariables() {
+  for (auto const& el : kGlobalVars) {
+    std::string key = el.first;
+    auto val = el.second;  // nested pairs. Clearer than el.second.second.
+    std::cout << key << " = " << val.first;
+    if (val.second != "")  // if description not empty
+      std::cout << " : " << val.second;
+    std::cout << "\n";
+  }
+}
+
+/**
+ * @brief Checks if key exists in kGlobalVars.
+ * @param key The key of the (key, value) pair.
+ * @return true, if key exists, false otherwise.
+ */
+bool IsSet(const std::string& key) {
```

Comment:
Please put the doxygen comments into the header file (above the declaration).

Reply:
done :)